### DISCUSSION TITLE
Is it possible to configure the tls ciphers used in daemon.json?

### DISCUSSION THREAD

**Discussion by rmontag-ap** [moby/moby#45121] [Q&A] [answered]
Hello,
we are using Docker CE 23.0.1 for our Jenkins installation on our worker nodes.
We have secured the docker daemon port (tcp:2375) with tls and client/server certificates.

We now got an incident from our security, that the daemon port is offering weak ciphers:

TLS 1.2 ciphers:
TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA
TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA

I checked the documentation (of the daemon.json) and found no hint, if and how the allowed tls ciphers can be configured; is it possible to configure them - and excluding the weak ciphers?

Greetings,
    Rainer

**Comment by cpuguy83** [comment, marked as answer]
Ciphers are hard-coded from here: https://github.com/docker/go-connections/blob/master/tlsconfig/config.go
I assume what you are seeing is from here: https://github.com/docker/go-connections/blob/58542c764a1173ea3dac965d89146c931a2946f7/tlsconfig/config.go#L44-L49

**Reply by cpuguy83** [reply]
I opened an issue to discuss removing the ciphers: https://github.com/docker/go-connections/issues/101
